### PR TITLE
totally disable publishing of non-core subprojects

### PR DESCRIPTION
**Pull request by SethTisue**
a with-extreme-prejudice followup to b21b2767

using code borrowed from scala/scala repo

### CODE REVIEW

**Comment by SethTisue** [comment]
this should avoid all the

```
java.lang.RuntimeException: Repository for publishing is not specified.
	at scala.sys.package$.error(package.scala:27)
	at sbt.Classpaths$$anonfun$getPublishTo$1.apply(Defaults.scala:1583)
	at sbt.Classpaths$$anonfun$getPublishTo$1.apply(Defaults.scala:1583)
```

stuff at e.g. https://travis-ci.org/scala/scala-parallel-collections/builds/213142948

File: build.sbt [hunk 1]
```diff
@@ -10,6 +10,17 @@ scalacOptions in ThisBuild      ++= Seq("-deprecation", "-feature")
...
+val disablePublishing = Seq[Setting[_]](
+  publishArtifact := false,
+  // The above is enough for Maven repos but it doesn't prevent publishing of ivy.xml files
+  publish := {},
+  publishLocal := {}
+)
+
+lazy val root = (project in file("."))
+  .settings(disablePublishing)
+  .aggregate(core, junit, scalacheck, testmacros)
```
Comment (dwijnand) [inline build.sbt]: Minor/pro-tip: You can avoid having to enumerate these, as well as maintain the enumeration, by just specifying the settings at the root level:

```scala
disablePublishing
```

File: build.sbt [hunk 2]
```diff
@@ -29,21 +40,20 @@ lazy val core = project.in(file("core")).settings(scalaModuleSettings).settings(
   )
 )
 
-lazy val junit = project.in(file("junit")).settings(
+lazy val junit = project.in(file("junit")).settings(disablePublishing).settings(
```
Comment (dwijnand) [inline build.sbt]: Minor/pro-tip: No need to jump in and out of `.settings`.

TL;DR: you could define this as

```scala
project.in(file("junit")).settings(
  disablePublishing,
  libraryDependencies += "com.novocode" % "junit-interface" % "0.11" % "test",
```

---

Long-winded explanation:

Individual settings (type `Setting[_]`) and seq of settings (type `Seq[Setting[_]]`) are both `SettingDefinition`s (the former by definition, the latter via implicit lift), and `.settings` is defined to take a varargs of `SettingDefinition`, so they can be defined next to each other.

For convenience there is also a `Def.settings(ss: SettingsDefinition*): Seq[Setting[_]]`

**Comment by SethTisue** [comment]
rebased, and applied Dale's suggested minor changes (thanks Dale)

**Comment by szeiger** [comment]
It would be great if sbt had support for this out of the box. Disabling publishing of a subproject is rather common and not at all well supported at the moment.

**Comment by dwijnand** [comment]
I agree. Between sbt, sbt-bintray, sbt-sonatype there's a lot of work to do just in this niche.

**Comment by SethTisue** [comment]
not to mention sbt-pgp, which is exactly where these particular exceptions were coming from (and are hopefully finally fixed by #20)

**Comment by SethTisue** [comment]
I closed and released the staging repo in the Sonatype web UI. I'll come back here once the artifacts are on Maven Central and I've done a bit of testing.

**Comment by SethTisue** [comment]
```
> update
[info] Updating {file:/Users/tisue/tmp/2.13-latest-nightly/}root-2-13-latest-nightly...
[info] Resolving jline#jline;2.14.3 ...
[info] downloading https://repo1.maven.org/maven2/org/scala-lang/modules/scala-parallel-collections_2.13.0-pre-e2a2cba/0.1.1/scala-parallel-collections_2.13.0-pre-e2a2cba-0.1.1.jar ...
[info] 	[SUCCESSFUL ] org.scala-lang.modules#scala-parallel-collections_2.13.0-pre-e2a2cba;0.1.1!scala-parallel-collections_2.13.0-pre-e2a2cba.jar(bundle) (372ms)
[info] Done updating.
[success] Total time: 2 s, completed Mar 21, 2017 5:41:28 PM
> console
[info] Starting scala interpreter...
[info] 
Welcome to Scala 2.13.0-pre-e2a2cba (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_112).
Type in expressions for evaluation. Or try :help.

scala> def foo = List(1).par.map(_ + 1).seq
<console>:11: error: value par is not a member of List[Int]
       def foo = List(1).par.map(_ + 1).seq
                         ^

scala> import scala.collection.parallel.CollectionConverters._
import scala.collection.parallel.CollectionConverters._

scala> def foo = List(1).par.map(_ + 1).seq
foo: scala.collection.immutable.Seq[Int]

scala> foo
res0: scala.collection.immutable.Seq[Int] = Vector(2)
```

🎉
